### PR TITLE
Fix warnings and errors in docs build

### PR DESCRIPTION
**Pull request by martyngigg**
Fixes the warnings & errors that are present in a build of the documentation.

**To test:**

Remove the `build/docs/html` & `build/docs/doctrees` directories and build the documentation. Build the html documentation and check that there are no errors. Note that you will need `MAKE_VATES=ON` to avoid warnings around `SaveWorkspaceToVTK`.

No issue number.

_Does not need to be in the release notes._

---
#### Reviewer

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):
##### Code Review
- [ ] Is the code of an acceptable quality?
- [ ] Does the code conform to the [coding standards](http://www.mantidproject.org/Coding_Standards)? Is it well structured with small focussed classes/methods/functions?
- [ ] Are there unit/system tests in place? Are the unit tests small and test the a class in isolation?
- [ ] If there are changes in the release notes then do they describe the changes appropriately?
##### Functional Tests
- [ ] Do changes function as described? Add comments below that describe the tests performed?
- [ ] How do the changes handle unexpected situations, e.g. bad input?
- [x] Has the relevant documentation been added/updated?
- [ ] Is user-facing documentation written in a user-friendly manner?
- [ ] Has developer documentation been updated if required?
- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

**Comment by AndreiSavici** [comment]
There are no warnings now. I've build the docs-qthelp instead.
I've set my Facility as `SNS`. I get several `IndirectDataReduction-[Warning] Failed to load instrument with error: Invalid value for property Filename (string) "/home/3y9/Mantid/mantid/instrument/VISION_Definition.xml": File "/home/3y9/Mantid/mantid/instrument/VISION_Definition.xml" not found. The current facility may not be fully supported.` and one `TomographyGUI-[Notice] Failed to initialize remote compute resource(s). This custom interface will not work. Error description: Failed to initialize because the facility is  SNS (and not ISIS). Facility not supported. This interface is designed to be used at ISIS. You will probably not be able to use it in a useful way because your facility is SNS. If you have set that facility by mistake in your settings, please update it.` messages. But they do not show up as sphinx warnings.

:shipit:

File: docs/source/algorithms/EnggFitPeaks-v1.rst [hunk 1]
```diff
@@ -36,14 +36,13 @@ the Mantid :ref:`ConvertUnits <algm-ConvertUnits>`. See also
...
-:ref:`Back2BackExponential <func-Back2BackExponential>`. Other
+:ref:`Back2BackExponential <func-BackToBackExponential>`. Other
 alternatives might be added as optional in the future (for example the
 simpler :ref:`Gaussian <func-Gaussian>` or the more complex
-:ref:`Bk2BkExpConvPV <func-Bk2BkExpConvPV` or :ref:`IkedaCarpenterPV
-<func-IkedaCarpenterPV>`). To produce an initial guess for the peak
-function parameters this algorithm uses the :ref:`FindPeaks
-<algm-FindPeaks>` algorithm starting from the expected peaks list
-given in the inputs.
+:ref:`Bk2BkExpConvPV <func-NeutronBk2BkExpConvPVoigt>` or :ref:`IkedaCarpenterPV <func-IkedaCarpenterPV>`). To
```
Comment (FedeMPouzols) [inline docs/source/algorithms/EnggFitPeaks-v1.rst]: I just caught this by chance. We have a bit of confusion around these "back to back exponential" functions. Thanks for fixing these links!
Bk2BkExpConvPV is one of the undocumented functions: https://github.com/mantidproject/mantid/issues/16480.
NeutronBk2BkExpConvPVoigt has different parameters and can be more confusing than helpful. Maybe it's better to leave the text `Bk2BkExpConvPV` without a link?

Comment (martyngigg) [inline docs/source/algorithms/EnggFitPeaks-v1.rst]: Good catch. I thought I read in a file that Bk2BkExpConvPV had just been renamed but I must have misunderstood.

I think you are correct, it should not have a link if one does not exist. I'll take it out.

**Comment by FedeMPouzols** [comment]
The rhel7 failure is totally unrelated to this.

**Pull request merged**